sdk/event/plugin_server.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package event
+
+import (
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+)
+
+// Serve is called from within a plugin and wraps the provided
+// Database implementation in a databasePluginRPCServer object and starts an
+// RPC server.
+func Serve(ev EventSubscriptionPlugin) {
+	plugin.Serve(ServeConfig(ev))
+}
+
+func ServeConfig(ev EventSubscriptionPlugin) *plugin.ServeConfig {
+	err := pluginutil.OptionallyEnableMlock()
+	if err != nil {
+		panic(err)
```

Comment:
Is there any reason we can't return an error from this function instead of panicking?

Reply:
I was mostly copying `sdk/database/dbplugin/server.go`, though that will return `nil`, which I think is more confusing.

It's easy enough to return an error, but we'd still probably immediately panic with it in `Serve`, since it is a function that is meant to never return.